containers/base/Dockerfile

```diff
@@ -116,27 +111,39 @@ RUN /tools/node/bin/npm install \
         node-cache@3.2.0 && \
     cd / && \
     /tools/node/bin/npm install -g forever && \
-    /tools/node/bin/npm install -g typescript
 
 # Install ProtoBuf, TensorFlow
-RUN pip install protobuf==3.0.0b2.post2 && \
+    pip install --no-cache-dir protobuf==3.0.0b2.post2 && \
     wget https://storage.googleapis.com/cloud-datalab/deploy/tf/tensorflow-0.11.0rc0-cp27-none-linux_x86_64.whl && \
-    pip install tensorflow-0.11.0rc0-cp27-none-linux_x86_64.whl && \
-    rm tensorflow-0.11.0rc0-cp27-none-linux_x86_64.whl
+    pip install --no-cache-dir tensorflow-0.11.0rc0-cp27-none-linux_x86_64.whl && \
+    rm tensorflow-0.11.0rc0-cp27-none-linux_x86_64.whl && \
 
 # Install DataFlow
-RUN pip install google-cloud-dataflow==0.4.2
+    pip install --no-cache-dir google-cloud-dataflow==0.4.2 && \
 
 # Install CloudML SDK
-RUN gsutil cp gs://cloud-ml/sdk/cloudml-0.1.6-alpha.tar.gz cloudml-0.1.6-alpha.tar.gz && \
-    pip install cloudml-0.1.6-alpha.tar.gz && \
-    rm cloudml-0.1.6-alpha.tar.gz
+    gsutil cp gs://cloud-ml/sdk/cloudml-0.1.6-alpha.tar.gz cloudml-0.1.6-alpha.tar.gz && \
+    pip install --no-cache-dir cloudml-0.1.6-alpha.tar.gz && \
+    rm cloudml-0.1.6-alpha.tar.gz && \
+
+# Clean up
+    apt-get purge -y build-essential bzip2 cpp cpp-4.9 python-setuptools pkg-config libfreetype6-dev && \
+    apt-get autoremove -y && \
+    rm -rf /var/lib/apt/lists/* && \
```

Comment:
What I meant is that they would now have to run "apt-get update" where they previously did not, and they would have to do so on every restart of Datalab.

To me, this feels like something we should bundle into the image, since it is a development tool and we want to favor optimizing the user's day-to-day experience over optimizing their getting-started experience.